types/base/api.d.ts

```diff
@@ -1,57 +1,75 @@
-type Part = {
-    name: string;
-    material: string;
-    color: string;
-}
-
-type InitialsExtra = Record<string, {
-    initials: string;
-    engraving?: string;
-}>
-
-type Spec = {
-    brand: string;
-    model: string;
-    parts: Part[];
-    initials: string;
-    engraving: string;
-    initialsExtra: InitialsExtra;
-    variant?: string;
-    version?: string;
-    description?: string;
-};
-
-type ImportOrderOptions = {
-    brand?: string;
-    model?: string;
-    variant?: string;
-    parts?: Part[];
-    initials?: string;
-    engraving?: string;
-    initialsExtra?: InitialsExtra;
-    gender?: string;
-    size?: number;
-    pending?: boolean;
-    notify?: boolean;
-    productId?: number;
-    currency?: string;
-    country?: string;
-    meta?: string[];
-}
-
-type RequestOptions = {
-    url?: string;
-    method?: string;
-    params?: Record<string, unknown>,
-    headers?: Record<string, unknown>,
-    auth?: boolean
-}
-
-export declare class RipeAPI {
-    key?: string;
-
-    constructor(options?: unknown);
-    authKeyP(key: string, options?: RequestOptions): Promise<void>;
-    importOrderP(ffOrderId: string, options?: ImportOrderOptions): Promise<void>;
-    _queryToSpec(query: string): Spec;
-}
+import { Order } from "../api";
+
+export type Part = {
+    name: string;
+    material: string;
+    color: string;
+    hidden?: boolean;
+    optional?: boolean;
+};
+
+export type InitialsExtra = Record<
+    string,
+    {
+        initials: string;
+        engraving?: string;
+    }
+>;
+
+export type Spec = {
+    brand: string;
+    model: string;
+    parts: Part[];
+    initials: string;
+    engraving: string;
+    initialsExtra: InitialsExtra;
+    variant?: string;
+    version?: string;
+    description?: string;
+};
+
+export type ImportOrderOptions = {
+    brand?: string;
+    model?: string;
+    variant?: string;
+    parts?: Part[];
+    initials?: string;
+    engraving?: string;
+    initialsExtra?: InitialsExtra;
+    gender?: string;
+    size?: number;
+    pending?: boolean;
+    notify?: boolean;
+    productId?: number;
+    currency?: string;
+    country?: string;
+    meta?: string[];
+};
+
+export type RequestOptions = {
+    url?: string;
+    method?: string;
+    params?: Record<string, unknown>;
+    headers?: Record<string, unknown>;
+    auth?: boolean;
+};
+
+export type GetRequestOptions = RequestOptions & {
+    params?: {
+        filters?: string[];
+        sort?: string[];
+        skip?: number;
+        limit?: number;
+    };
+};
+
+export declare class RipeAPI {
+    key?: string;
+
+    constructor(options?: unknown);
+    authKeyP(key: string, options?: RequestOptions): Promise<void>;
+    importOrderP(ffOrderId: string, options?: ImportOrderOptions): Promise<Order>;
+    getOrdersP(options?: GetRequestOptions): Promise<Order[]>;
+    deleteOrderP(number: number, options?: RequestOptions): Promise<XMLHttpRequest | null>;

```

Comment:
Even though it's true that it returns `{ success: true }` or something like that, that's just a coincidence and has no semantic meaning so the type should be null.

Reply:
But the return sequence is the following:

- https://github.com/ripe-tech/ripe-sdk/blob/master/src/js/api/order.js#L129
- https://github.com/ripe-tech/ripe-sdk/blob/master/src/js/api/order.js#L109
- https://github.com/ripe-tech/ripe-sdk/blob/29fafa0b52866dd383d868491a20752282e3ebd3/src/js/base/api.js#L266 OR  https://github.com/ripe-tech/ripe-sdk/blob/29fafa0b52866dd383d868491a20752282e3ebd3/src/js/base/api.js#L271
- From there it's going to be either `null` or the result of `_requestURLFetch` or `_requestURLLegacy`
- `_requestURLFetch` seems null return as well, so its `null` or `_requestURLLegacy`
- `_requestURLLegacy` returns a `const request = new XMLHttpRequest();`
- So seems like `XMLHttpRequest | null` no?

Reply:
Ah, so you are saying that the `XMLHttpRequest` is returned so that we can await it but it's irrelevant as a type?

Reply:
yes